### PR TITLE
new: Google Scholar/Highwire metadata

### PR DESCRIPTION
**Pull request by gabestein**
This generates Google/Highwire meta tags for references on the frontend for pubs that use built-in footnotes or citations. It's a pretty naive generator; on the other hand, the tags themselves are [not well specified](https://citation.js.org/blog/?post=3781582023912504297). Eventually, we should probably write a CSL to produce these, but that seems like a multi-week project not particularly worth our time at the moment.

Features of this particular PR:
- De-duplicates among footnotes and tags
- Supports journal, conference, and book (defaulting to journal)
- Filters out unstructured inputs

_Test plan_
1. Visit pub with lots of citations and verify they are represented in meta tags 
2. Visit pub with duplicate tags (both within footnote/citation and across footnote/citation) and verify they are _not_ replicated in meta tags. (ex: https://hdsr.duqduq.org/pub/hx3pgxjc)
3. Visit pub with journal, book, and conference citations and verify they are represented properly with `citation_journal_title`, `citation_book_title`, and `citation_conference_title` respectively.
4. Create new pub. Publish with no references. Then add some references and publish. Make sure meta tags are not there before, there after, and that merging works as expected.

### CODE REVIEW

File: shared/pub/metadata.js [hunk 1]
```diff
@@ -49,3 +49,49 @@ export const getTextAbstract = (docJson) => {
 	}
 	return abstract;
 };
+
+export const getGSNotes = (notes) => {
```
Comment (idreyn) [inline shared/pub/metadata.js]: Could we call this `getGoogleScholarNotes` instead? `GS` is a little opaque for someone reading this code out of context.

File: shared/pub/metadata.js [hunk 2]
```diff
@@ -49,3 +49,49 @@ export const getTextAbstract = (docJson) => {
...
+	return notes.reduce((unique, note) => {
+		const noteArray = [];
+		const noteType = note.json[0].type;
+		const noteTypes = {
```
Comment (idreyn) [inline shared/pub/metadata.js]: Nit: this can be factored out of the function.

File: shared/pub/metadata.js [hunk 3]
```diff
@@ -49,3 +49,49 @@ export const getTextAbstract = (docJson) => {
...
+			chapter: 'book',
+			book: 'book',
+			proceedings: 'conference',
+			'paper-conference': 'conference',
+		};
+		const noteTypeString = noteTypes[noteType] || 'journal';
+
+		const noteArr = Object.entries(note.json[0]);
+		noteArr.forEach((entry) => {
```
Comment (idreyn) [inline shared/pub/metadata.js]: Personally I'd find this a bit easier to read as `Object.entries(...).forEach(([key, value]) => { ... })`. Something about `entry[0]` or `entry[1]` just adds cognitive overhead in remembering what it is I'm looking at. And having `noteArray` and `noteArr` in the same scope is a little confusing too.

Comment (gabestein) [reply shared/pub/metadata.js]: Done, thanks. Had no idea you could do that in forEach (is that...destructuring, what is that?). Anyhow, super helpful.

File: shared/pub/metadata.js (hunk 1 above)
Comment (gabestein) [reply shared/pub/metadata.js]: +1

File: shared/pub/metadata.js (hunk 2 above)
Comment (gabestein) [reply shared/pub/metadata.js]: I think this is what you meant?

**Pull request merged**